src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -170,18 +171,25 @@ def _make_compile_request(self, versioned_target, dependencies, external_libs_pr
     include_dirs.extend(self._get_third_party_include_dirs(external_libs_product, dependencies))
 
     sources_and_headers = self.get_sources_headers_for_target(target)
-
     return NativeCompileRequest(
       compiler=self._compiler,
       include_dirs=include_dirs,
       sources=sources_and_headers,
-      fatal_warnings=self._compile_settings.get_fatal_warnings_value_for_target(target),
+      fatal_warnings=(self._compile_settings
+                         ._native_build_step_settings
+                         .get_fatal_warnings_value_for_target(target)),
+      compiler_options=(self._compile_settings
+                           ._native_build_step_settings
+                           .get_merged_compiler_options_for_target(target,
```

Comment:
It looks like the `get_merged_compiler_options_for_target` method already includes the enabled/disabled args dicts... are the arguments here a different collection of options? If so, is there a usecase for that? It's sortof confusing.